src/VirtoCommerce.Platform.Core/Jobs/RecurringJobExtensions.cs

```diff
@@ -0,0 +1,121 @@
+        public static void WatchJobSetting<T>(this IRecurringJobManager recurringJobManager,
+            ISettingsManager settingsManager,
+            SettingDescriptor enablerSetting,
+            SettingDescriptor cronSetting,
+            Expression<Func<T, Task>> methodCall,
+            string jobId,
+            TimeZoneInfo timeZoneInfo,
+            string queue)
+        {
+            if (recurringJobManager == null) throw new ArgumentNullException(nameof(recurringJobManager));
+            if (settingsManager == null) throw new ArgumentNullException(nameof(settingsManager));
+
+            var settingCronJob = new SettingCronJobBuilder(new SettingCronJob())
+                .SetEnablerSetting(enablerSetting)
+                .SetCronSetting(cronSetting)
+                .SetJobId(jobId)
+                .SetQueueName(queue)
+                .SetTimeZoneInfo(timeZoneInfo)
+                .ToJob(methodCall)
+                .Build();
+
+            _observedSettingsDict.AddOrUpdate(settingCronJob.EnableSetting.Name, settingCronJob, (settingName, сronJob) => settingCronJob);
+
+            _observedSettingsDict.AddOrUpdate(settingCronJob.CronSetting.Name, settingCronJob, (settingName, сronJob) => settingCronJob);
+
+            RunOrRemoveJobAsync(recurringJobManager, settingsManager, settingCronJob).GetAwaiter().GetResult();
+        }
+        
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="recurringJobManager"></param>
+        /// <param name="settingsManager"></param>
+        /// <param name="message"></param>
+        /// <returns></returns>
+        public static async Task HandleSettingChange(this IRecurringJobManager recurringJobManager, ISettingsManager settingsManager, ObjectSettingChangedEvent message)
+        {
+            if (recurringJobManager == null) throw new ArgumentNullException(nameof(recurringJobManager));
+            if (settingsManager == null) throw new ArgumentNullException(nameof(settingsManager));
+
+            foreach (var changedEntry in message.ChangedEntries.Where(x => x.EntryState == EntryState.Modified
+                                              || x.EntryState == EntryState.Added))
+            {
+                if (_observedSettingsDict.TryGetValue(changedEntry.NewEntry.Name, out var settingCronJob))
+                {
+                    await RunOrRemoveJobAsync(recurringJobManager, settingsManager, settingCronJob);
+                }
+            }
+        }
+
+
+        private static async Task RunOrRemoveJobAsync(this IRecurringJobManager recurringJobManager, ISettingsManager settingsManager, SettingCronJob settingCronJob)
+        {
+            var processJobEnable = await settingsManager.GetValueAsync(settingCronJob.EnableSetting.Name, (bool)settingCronJob.EnableSetting.DefaultValue);
+            if (processJobEnable)
+            {
+                recurringJobManager.RemoveIfExists(settingCronJob.RecurringJobId);
```

Comment:
Seems `RemoveIfExists` could be done before the `if` statement, so `else` would not be needed at all.

Reply:
remove